### PR TITLE
Saint Game CheatSheet

### PR DESCRIPTION
**Pull request by jatinparmar96**
What does your Instant Answer do?

Shows all the cheat codes for saints row 4 game.

What is the data source for your Instant Answer? (Provide a link if possible)

http://www.cheatcc.com/ps3/saintsrow4cheatscodes.html

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

Gamers.

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)

screenshot 2

---
https://duck.co/ia/view/saints_row_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@jatinparmar96 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by mintsoft** [comment]
@jatinparmar96 Can you the link to the IA page into your PR?

**Comment by MariagraziaAlastra** [comment]
Hey @jatinparmar96, thanks for contributing!
Did you know there's no need to fill in the PR template anymore? 
Just create your IA Page [here](https://duck.co/ia/new_ia) and then paste the link to it in your first comment, as @mintsoft suggested.
You can find more info [in the docs](http://docs.duckduckhack.com/submitting/pull-request.html#submitting-a-pull-request).
Thanks!

**Comment by jatinparmar96** [comment]
Thanks @mintsoft and @MariagraziaAlastra for the info. I have include the link to the IA page.

**Comment by moollaza** [comment]
@jatinparmar96 Thanks for contributing and welcome to DuckDuckHack!

Currently, all our cheat sheets have `cheat_sheet`in the ID. I've gone ahead and changed your IA Pages ID and updated the link in your PR.

Please update the JSON to reflect the new ID: `saints_row_cheat_sheet`. Please change the filename to `saints-row.json`

File: share/goodie/cheat_sheets/json/saint.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
+{  
+   "id":"saints",
+   "name":"Saints Row IV",
+   "description":"Pause the game, and select the 'Extras' option, then choose the 'Cheats' selection. Select 'Add Cheat' and enter one of the following codes to activate the corresponding cheat function. If you entered the code correctly, you will hear a sound. Note: Trophies and auto save are disabled when cheats are activated. However, you are still allowed to manually save while cheats are enabled.",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/saint.json]: This is too long. It should describe the cheat sheet. I think you should break these steps down into a new section in the Cheat Sheet

File: share/goodie/cheat_sheets/json/saint.json [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+   "metadata":{  
+      "sourceName":"Saints Row IV",
+      "sourceUrl":"http://www.cheatcc.com/ps3/saintsrow4cheatscodes.html"
+   },
+   "aliases":[  
+      "saints row 4"
+   ],
+   "template_type":"keyboard",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/saint.json]: Let's use the `reference` template -- we're not providing keyboard keys in here :+1:

**Comment by jatinparmar96** [comment]
My json file is valid then why is this test failing??

File: share/goodie/cheat_sheets/json/saints-row.json [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      "sourceUrl":"http://www.cheatcc.com/ps3/saintsrow4cheatscodes.html"
+   },
+   "aliases":[  
+      "saints row 4","sr 4"
+   ],
+   "template_type":"reference",
+   "section_order":[  
+      "Player Ability",
+      "World",
+      "All Unlockables"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/saints-row.json]: Looks like you forgot `Vehicles` -- please add that in :+1:

**Comment by moollaza** [comment]
Thanks! This LGTM :+1:

**Comment by moollaza** [comment]
@jatinparmar96 this LGTM :+1:

Thanks a lot for contributing (we really appreciate it :smiley:). We invite you to stick around and comment/help on other PR's & Issues or even submit an Instant Answer!

**Comment by moollaza** [comment]
@jatinparmar96 thanks again for creating the Saints Row Cheat Sheet! You did a great job. We at DuckDuckGo, the DuckDuckHack community, and all the people who will encounter this Instant Answer really appreciate it.

We could really use your help on something else now. We want to increase our Instant Answer coverage for as many programming languages and platforms as possible. We want an AppleScript Cheat Sheet and I think you'd similarly do a great job with building that. Please update the [Programming IA Coverage table](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage) (mark it with a 'd') if you'd be willing to take it up!

I think it is a good fit for you, but I'm of course not entirely sure, so I'd love to chat if you're around this week or next. Please message me on slack (@moollaza) or send me an email (moollaza@duckduckgo.com) if you'd be interested in talking. 

Thanks again!

**Pull request merged**